src/rules/no-named-default.js

```diff
@@ -0,0 +1,32 @@
+module.exports = {
+  meta: {
+    docs: {},
+  },
+
+  create: function (context) {
+    function checkSpecifiers(type, node) {
+      if (node.source == null) return
+
+      const hasImportSpecifier = node.specifiers.some(function (im) {
+        return im.type === type
+      })
+
+      if (!hasImportSpecifier) {
+        return
+      }
+
+      node.specifiers.forEach(function (im) {
+        if (im.type !== type) return
+
+        if (im.imported.name === 'default') {
```

Comment:
I also went ahead and removed the preceding checks as I don't believe they were helping us anymore.